src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -234,6 +234,17 @@ private Type0CommandMessageSection<BsonDocument> CreateType0Section(ConnectionDe
             return new Type0CommandMessageSection<BsonDocument>(_command, elementAppendingSerializer);
         }
 
+        private bool IsRetryableWriteExceptionAndDeploymentDoesNotSupportRetryableWrites(MongoCommandException exception)
+        {
+            return
+                exception.Result.TryGetValue("code", out var errorCode) &&
+                errorCode.ToInt32() == 20 &&
+                exception.Result.TryGetValue("errmsg", out var errmsg) &&
+                errmsg.IsString &&
+                errmsg.ToString().StartsWith("Transaction numbers");
```

Comment:
It probably would have been safe to assume that errmsg is always a string and just use errmsg.AsString

Reply:
Fixed!